activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/test/java/org/activiti/cloud/services/query/events/handlers/ProcessCandidateStarterGroupAddedEventHandlerTest.java

```diff
@@ -58,6 +61,27 @@ public void handleShouldStoreNewProcessCandidateGroupUser() {
         assertThat(captor.getValue().getGroupId()).isEqualTo(event.getEntity().getGroupId());
     }
 
+    @Test
+    public void handleShouldNotStoreProcessCandidateGroupIfExists() {
+        //given
+        ProcessCandidateStarterGroupImpl candidateGroup = new ProcessCandidateStarterGroupImpl(UUID.randomUUID().toString(),
+                                                                                               UUID.randomUUID().toString());
+        CloudProcessCandidateStarterGroupAddedEventImpl event = new CloudProcessCandidateStarterGroupAddedEventImpl(candidateGroup);
+
+        //when
+        handler.handle(event);
+        when(entityManager.find(ProcessCandidateStarterGroupEntity.class,
+                                new ProcessCandidateStarterGroupId(candidateGroup.getProcessDefinitionId(), candidateGroup.getGroupId())))
+            .thenReturn(new ProcessCandidateStarterGroupEntity());
+        handler.handle(event);
```

Comment:
Suggestion: as this is a unit test and the case where handle is storing the entity is already tested in the previous test, I think you can remove the call to `handle.handle(event)` at the line `72` and keep only the one at the line `75`. But you can keep it like this if you think it's clearer on that way.